clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/SystemElasticsearchDAO.scala

```diff
@@ -0,0 +1,111 @@
+package org.broadinstitute.clio.server.dataaccess
+
+import akka.pattern.after
+import org.broadinstitute.clio.server.ClioServerConfig
+import org.broadinstitute.clio.server.dataaccess.elasticsearch.ElasticsearchIndex
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+trait SystemElasticsearchDAO extends SearchDAO { this: HttpElasticsearchDAO =>
```

Comment:
Hopefully discussed in person, still:

> It allows SystemElasticsearchDAO to call getClusterHealth and so on?

Yes, it specifies that an SystemElasticsearchDAO may only be used in a place where the trait is mixed in with an HttpElasticsearchDAO.

Meanwhile, HttpElasticsearchDAO mixes in SystemElasticsearchDAO, so the contract is satisfied.

> How did you arrive at this design?

I arrived at this pattern because I wanted to start separating different parts of a massive HttpElasticsearchDAO into separate traits. The underlying SearchDAO interface/trait will eventually contain dozens of abstract methods. Instead of having all the concrete methods implemented in one massive class, I'd rather have them grouped into different mixins.

H/T also to the Firecloud-Orchestration [design](https://broadinstitute.atlassian.net/wiki/spaces/DSDEGP/pages/114531509/Clio+Metadata+Manager#Clio(MetadataManager)-PackageStructureandSingleResponsibilities).